randovania/server/multiplayer/session_admin.py

```diff
@@ -217,6 +220,13 @@ def _update_layout_generation(sa: ServerApp, session: MultiplayerSession, world_
         session.save()
 
 
+def _check_preset_consistency(presets: list[VersionedPreset], session: MultiplayerSession) -> None:
+    unsupported_games = {preset.game.data.long_name for preset in presets if preset.game not in session.allowed_games}
```

Comment:
if using the above suggested aproach, then this function would simply be able to call the first `get_unsupported_games`  function

Reply:
Nah, you are mixing backend and frontend code.
Your suggestion was about frontend but this part here is the backend.

Reply:
whoops, didnt notice. thanks